metaflow/plugins/aip/skip_decorator.py

```diff
@@ -0,0 +1,52 @@
+# Skip decorator is a workaround solution to implement conditional branching in metaflow.
+# When condition variable is_skipping is evaluated to True,
+# it will skip current step and execute the supplied next step.
+
+from functools import wraps
+from metaflow.decorators import StepDecorator
+
+
+class SkipDecorator(StepDecorator):
+    """
+    The @skip decorator is a workaround for conditional branching. The @skip decorator checks an artifact
+    and if it is false, skips the evaluation of the step function and jumps to the supplied next step.
+
+    **The `start` and `end` steps are always expected and should not be skipped.**
+
+    Usage:
+        class SkipFlow(FlowSpec):
+
+        condition = Parameter("condition", default=False)
+
+        @step
+        def start(self):
+            print("Should skip:", self.condition)
+            self.next(self.middle)
+
+        @skip(check='condition', next='end')
+        @step
+        def middle(self):
+            print("Running the middle step - not skipping")
+            self.next(self.end)
+
+        @step
+        def end(self):
+            pass
+    """
+
+    name = "skip"
+
+    def __init__(self, check="", next=""):
+        super().__init__()
+        self.check = check
+        self.next = next
```

Comment:
The feature ask is to skip a step.
Introducing conditional branching (a conditional `next` step -> conditional branching) is out of scope.

There are valid deep reasons to not allow conditional branching; the mainly that upsteam Metaflow OSS has not implemented this feature because conditional branching introduces for complex DAG reasoning and branching that is difficult to test and reason over for the applied scientist.  For example, this next conditional could goto any step.

Reply:
This is the version that RMX is using right now: https://gitlab.zgtools.net/analytics/artificial-intelligence/rmx/workflows/rmx-delg-eventing/-/merge_requests/43/diffs

It's also the same version provided by Ville in https://zillowgroup.slack.com/archives/C02116BBNTU/p1660145168960519?thread_ts=1660134373.601939&cid=C02116BBNTU